src/mono/mono/metadata/class.c

```diff
@@ -372,8 +372,12 @@ mono_type_get_name_recurse (MonoType *type, GString *str, gboolean is_recursed,
 		g_string_append_c (str, '[');
 		if (rank == 1)
 			g_string_append_c (str, '*');
-		for (i = 1; i < rank; i++)
-			g_string_append_c (str, ',');
+		else if (rank > 64)
+			// Only taken in an error path, runtime will not load arrays of more than 32 dimentions
+			g_string_printf (str, "%d", rank);
```

Comment:
Did you mean to use `g_string_append_c` here? `g_string_printf` is going to free the old string and replace it with your inputted one, which doesn't seem right looking at your comment here https://github.com/dotnet/runtime/pull/36604#issuecomment-629818957

Otherwise, looks fine from the Mono side of things—thanks for this!

Reply:
I do need to print an integer though - `g_string_append_c` seems to only support printing a char?

Reply:
Maybe `g_string_append_printf`?

Reply:
Yeah I edited my comment right after posting when I realized my mistake there, sorry 😄 
Looks fine now, thanks again!